src/main/java/it/polimi/telco/model/ServicePackage.java

```diff
@@ -0,0 +1,39 @@
+package it.polimi.telco.model;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.List;
+
+@Entity
+@Table(name = "service_package", schema = "telco")
+@NamedQuery(name = "ServicePackage.getServicePackageById", query = "SELECT sp FROM ServicePackage sp  WHERE sp.id = ?1 ")
+public class ServicePackage {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotNull
+    private String name;
+
```

Comment:
_"A given service package is offered with different validity periods. The user can choose one of the offered validity periods among those associated with the package."_

We should associate service packages with some validity periods and then during the subscription, we associate one of those periods to the subscription